jvm/ml4ir-inference/src/test/integration/ml4ir/inference/tensorflow/TensorFlowInferenceIT.scala

```diff
@@ -0,0 +1,109 @@
+package ml4ir.inference.tensorflow
+
+import scala.collection.JavaConverters._
+import ml4ir.inference.tensorflow.data.{ModelFeaturesConfig, StringMapExampleBuilder, TestData}
+import org.junit.Test
+import org.junit.Assert._
+import org.tensorflow.example._
+
+import scala.collection.mutable.ListBuffer
+import scala.io.Source
+
+@Test
+class TensorFlowInferenceIT extends TestData {
+  val classLoader = getClass.getClassLoader
+
+  def resourceFor(path: String) = classLoader.getResource(path).getPath
+
+  /**
+   * Class holding a Example and prediction from Python
+   * @param sequence the Example context
+   * @param scores the prediction scores
+   */
+  class PredictionVector(var sequence: Map[String, String], var scores: Array[Float])
+
+  /**
+   * Ugly parsing of the Python CSV predicition.
+   *
+   * Mainly removing the "b'" formatting and slicing into List and Map.
+   *
+   * @param line
+   * @return
+   */
+  def extractColumnValues(line: String): PredictionVector = {
+    val cols = line.split(",").map(_.trim)
+
+    val sequence = Map("query_text" -> cols(1).substring(2, cols(1).length - 1),
+      "domain_id" -> cols(3).substring(2, cols(3).length - 1),
+      "user_context" -> cols(4).substring(2, cols(4).length - 2).filterNot("'b".toSet).trim().replace(" ", ","))
+
+    try {
+      val scores = cols(7).substring(2, cols(7).length - 2).split(" ").map(_.trim).map((s: String) => s.toFloat)
+      new PredictionVector(sequence, scores)
+    } catch {
+      case _: NumberFormatException => null
+    }
+
+  }
+
+  /**
+   * Read a prediction file from the python project.
+   * The prediction csv is not 1 row per line, so we need to reconcile the file.
```

Comment:
what do you mean by "not 1 row per line"?

Reply:
the 8 columns values of this CSV will be split in a variable number of lines... 

This is the first prediction_vector for example (4 lines in the csv output from python out of the box)

```
b'query_id_0',b'bernardo and unfold',"[b'bernardo' b'and' b'unfold' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b''
b'' b'' b'' b'' b'' b'']",b'S',"[b'DDD' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b''
b'' b'' b'']",b'canary',[0. 0. 1. 0. 0. 0. 0. 0. 0.],"[2.9891068e-01 2.4615353e-01 2.2974105e-01 8.6946599e-02 1.3798673e-01
```

Reply:
that's not a useful format.  Ugh.  We need to fix that.